code/modules/mob/living/carbon/human/death.dm

```diff
@@ -14,7 +14,12 @@
 			var/atom/movable/thing = I.remove(src)
 			if(thing)
 				thing.forceMove(get_turf(src))
-				thing.throw_at(get_edge_target_turf(src, pick(GLOB.alldirs)), rand(1,3), 5)
+				thing.throw_at(get_edge_target_turf(src, pick(GLOB.alldirs)), rand(1, 3), 5)
+
+	for(var/obj/item/I in get_equipped_items(include_pockets = TRUE))
+		unEquip(I)
```

Comment:
```suggestion
		unEquip(I, TRUE)
```
If you want to force it to unEquip. Else you can do a check around it to see if it gets unEquiped, if so throw it etc.

Reply:
Done